### ISSUE TITLE
Enable openstack driver can specify machine flavor otherwise the flavor id & specify image file otherwise imageid

### ISSUE THREAD

**Issue by ldpc** [docker/machine#910]
When use Openstack driver, we must specify the flavor-id and image-id.That's means the end-user must use Openstack api to create image and flavor.If we enable user can just only specify machine flavor and the url where we can download the image, that's make the Openstack driver more userable.

**Comment by ggiamarchi** [comment]
For images we can add a flag `--openstack-image-url`. But usally we don't want to create a new image each time we create a machine. In my opinion this option should always be used with `--openstack-image-name`. With this configuration docker-machine will look for an image matching the requested name. If no image is found, it will create one based on the given URL.

For flavors, i'm not sure what exactly you want to achieve. Do you want to give the amount vCPU, RAM and disk size to docker-machine instead of the flavor name or id ? If so, i think this won't be useful as long as only administrator are allowed to create flavors.

**Comment by Fangfenghua** [comment]
@ggiamarchi   I agree with you the about  the image, use the image name and the URL.  Also we can specify a meaningful names such as docker_machine_image. 
    For the flavor,I mean give the amount vCPU,RAM,DISK.For the a administrator  it's useful.Also, I suggest add a flavor name, so we can specify a meaningful names.

**Comment by ggiamarchi** [comment]
@Fangfenghua Actually the flag `--openstack-flavor-name` is already available. I just realized that it is not documented. The same for `--openstack-image-name`.

**Comment by Fangfenghua** [comment]
@ggiamarchi Great，so we  just need add the URL.

**Comment by Fangfenghua** [comment]
@ggiamarchi  Did you have any plane to add the --openstack-image-url, if you didn't you can assign it to me! I will do it!

**Comment by ggiamarchi** [comment]
@Fangfenghua I haven't planed to work on this one. Feel free to submit a PR.

**Comment by fsoppelsa** [comment]
@ggiamarchi @Fangfenghua  to enable `--image-image-url`, using Rackspace gophercloud, a Glance API is needed, but https://github.com/rackspace/gophercloud/issues/213.

The implementation of *compute/v2*, at the day, doesn't allow to create images (if not from snapshots). And `--copy-from` is deprecated in Glance v2.

@Fangfenghua do you still have ideas?
